lib/animation.js

```diff
@@ -210,6 +226,65 @@ Animation.prototype.speed = function(speed) {
 };
 
 /**
+ * This function is called in each frame of our animation
+ * Users need not call this. It's automatic
+ */
+
+ Animation.prototype.loopFunction = function(loop) {
+
+  // Find the current timeline progress
+  var progress = this.calculateProgress(loop.calledAt);
+
+  // Find the left and right cuePoints/keyFrames;
+  var indices = this.findIndices(progress);
+
+  // Get tweened value
+  var val = this.tweenedValue(indices, progress);
+
+  // call render function
+  this.target[Animation.render](val);
+
+  // If this animation has been running for more than 5 seconds
+  if (loop.calledAt > this.fallBackTime) {
+    this.fallBackTime = Infinity;
+    this.playLoop.stop();
+    this.playLoop = new temporalFallback(this);
+  }
+
+  // See if we have reached the end of the animation
+  if ((progress === 1 && !this.reverse) || (progress === this.loopback && this.reverse)) {
+
+    if (this.loop || (this.metronomic && !this.reverse)) {
+
+      if (this.onloop) {
+        this.onloop();
+      }
+
+      if (this.metronomic) {
+        this.reverse = this.reverse ? false : true;
+      }
+
+      this.normalizedKeyFrames = __.cloneDeep(this.keyFrames);
+      this.normalizedKeyFrames = this.target[Animation.normalize](this.normalizedKeyFrames);
+      this.normalizedKeyFrames = this.normalizeKeyframes();
```

Comment:
I had it in the back of my mind that users might want to use animation's normalize routine w/o the device's normalize routine though now that time has passed I can't imagine why.

I'll rework it.